### PR TITLE
fix: hide internal transaction origins in confirmation views

### LINKED ISSUE #11919: Hide origin in transaction confirmation if origin is internal
### What is this about?

Ensure transaction confirmations do not display the `origin` of the transaction metadata if the transaction was generated from the wallet.

### Scenario

_No response_

### Design

_No response_

### Technical Details

- Update `TransactionHeader` component.
- Consider handling all internal origins in mobile, currently `TransactionTypes.MMM`, `process.env.FOX_CODE` and `ORIGIN_METAMASK`.
- Verify no other origin display components.

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**
This PR modifies the transaction confirmation flow to hide the origin display when transactions are generated internally by the wallet.
**Changes**
- Updated `TransactionHeader` and  `ApproveTransactionHeader` components to conditionally render origin information
- Added logic to identify internal origins.

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/11919

## **Manual testing steps**

**Case 1**
1. Import a QR account
2. Go to test dapp
3. Perform a signature or transaction
4. and click `Confirm`

**Case 2**
1. Normal account , go to the test-dapp
2. Click `Set Approve for All`

### **Before**

before

### **After**

after

[origin pill.webm](https://github.com/user-attachments/assets/b15f2732-efc6-40cb-8a94-7dcae1e1b20e)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/TransactionHeader/index.test.tsx [hunk 1]
```diff
@@ -1,27 +1,81 @@
...
+      { state: mockInitialState },
     );
     expect(wrapper).toMatchSnapshot();
   });
+
+  it.each([
+    ['ORIGIN_METAMASK', ORIGIN_METAMASK],
+    ['MM_FOX_CODE', process.env.MM_FOX_CODE],
+    ['MMM', TransactionTypes.MMM],
+  ])('does not render if origin is internal (%s)', (_, origin) => {
```
Comment (matthewwalsh0) [inline app/components/UI/TransactionHeader/index.test.tsx:70]:
```suggestion
  ])('does not render origin if %s', (_, origin) => {
```

File: app/components/UI/TransactionHeader/index.js [hunk 2]
```diff
@@ -17,9 +17,12 @@ import {
   selectNickname,
   selectProviderType,
 } from '../../../selectors/networkController';
+import { INTERNAL_ORIGINS } from '../../../constants/transaction';
 
 const { ORIGIN_DEEPLINK, ORIGIN_QR_CODE } = AppConstants.DEEPLINKS;
 
+export const TRANSACTION_HEADER_ORIGIN = 'TRANSACTION_HEADER_ORIGIN';
```
Comment (matthewwalsh0) [inline app/components/UI/TransactionHeader/index.js:24]: Minor, is it standard in mobile to define this in a `e2e/selectors` file and import them?

File: app/components/Views/confirmations/components/ApproveTransactionHeader/ApproveTransactionHeader.tsx [hunk 3]
```diff
@@ -69,9 +70,11 @@ const ApproveTransactionHeader = ({
 
   const accountTypeLabel = getLabelTextByAddress(activeAddress) ?? undefined;
 
+  const showOrigin = !INTERNAL_ORIGINS.includes(origin) && origin && !isOriginDeepLink;
```
Comment (matthewwalsh0) [inline app/components/Views/confirmations/components/ApproveTransactionHeader/ApproveTransactionHeader.tsx:73]: Minor, is it more logical to put the `origin` condition first?

File: app/components/Views/confirmations/components/ApproveTransactionHeader/ApproveTransactionHeader.test.tsx [hunk 4]
```diff
@@ -154,4 +166,20 @@ describe('ApproveTransactionHeader', () => {
     const originPill = queryByTestId(APPROVAL_TAG_URL_ORIGIN_PILL);
     expect(originPill).toBeNull();
   });
+
+  it.each([
+    ['ORIGIN_METAMASK', ORIGIN_METAMASK],
+    ['MM_FOX_CODE', process.env.MM_FOX_CODE],
+    ['MMM', TransactionTypes.MMM],
+  ])('does not render if origin is internal (%s)', (_, origin) => {
```
Comment (matthewwalsh0) [inline app/components/Views/confirmations/components/ApproveTransactionHeader/ApproveTransactionHeader.test.tsx:174]:
```suggestion
  ])('does not render origin if %s', (_, origin) => {
```

**Review by matthewwalsh0 — DISMISSED** [review]

File: app/components/UI/TransactionHeader/index.js (hunk 2 above)
Comment (vinistevam) [reply app/components/UI/TransactionHeader/index.js]: Done https://github.com/MetaMask/metamask-mobile/pull/12100/commits/45b380b3582d8bae4ced900a37e6aeec4dd99272

File: app/components/Views/confirmations/components/ApproveTransactionHeader/ApproveTransactionHeader.tsx (hunk 3 above)
Comment (vinistevam) [reply app/components/Views/confirmations/components/ApproveTransactionHeader/ApproveTransactionHeader.tsx]: Done https://github.com/MetaMask/metamask-mobile/pull/12100/commits/45b380b3582d8bae4ced900a37e6aeec4dd99272

**Review by matthewwalsh0 — APPROVED** [review]

**Review by digiwand — APPROVED** [review]

**Pull request merged**